recipes-ni/ni-systemimage/files/nisystemimage

```diff
@@ -784,9 +843,11 @@ trap 'handle_err ${BASH_SOURCE} ${LINENO} ${FUNCNAME:-unknown} $?' ERR
 [[ -f $SCRIPT_ROOTFS ]] && . "$SCRIPT_ROOTFS"
 [[ -f $SCRIPT_PRE ]] && . "$SCRIPT_PRE"
 
-case "$(runlevel)" in
-*\ 6|*\ 0) die ILLTIMED "Called during shutdown; aborting." ;;
-esac
+if ! in_restoremode; then
+	case "$(runlevel)" in
+	*\ 6|*\ 0) die ILLTIMED "Called during shutdown; aborting." ;;
+	esac
+fi
```

Comment:
Why would we not want to do this runlevel check in restoremode?

Reply:
We use our own `/init` in restoremode and not the one from sysvinit so `runlevel` returns `unknown`.